Tools/AP_Periph/networking_passthru.cpp

```diff
@@ -0,0 +1,135 @@
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("EP1", 2,  Networking_Periph::Passthru, ep1, -1),
+
+    // @Param: EP2
+    // @DisplayName: Endpoint 2
+    // @Description: Passthrough Endpoint 2. This can be a serial port UART, a Network port, or a CAN port. The selected port will route to Endport 1.
+    // @CopyFieldsFrom: NET_PASS1_EP1
+    AP_GROUPINFO("EP2", 3,  Networking_Periph::Passthru, ep2, -1),
+
+    // @Param: BAUD1
+    // @DisplayName: Endpoint 1 Baud Rate
+    // @Description: The baud rate used for Endpoint 1. Only applies to serial ports.
+    // @CopyFieldsFrom: SERIAL1_BAUD
+    AP_GROUPINFO("BAUD1", 4,  Networking_Periph::Passthru, baud1, 115200),
+
+    // @Param: BAUD2
+    // @DisplayName: Endpoint 2 Baud Rate
+    // @Description: The baud rate used for Endpoint 2. Only applies to serial ports.
+    // @CopyFieldsFrom: SERIAL1_BAUD
+    AP_GROUPINFO("BAUD2", 5,  Networking_Periph::Passthru, baud2, 115200),
+
+    // @Param: OPT1
+    // @DisplayName: Serial Port Options EP1
+    // @Description: Control over UART options for Endpoint 1. Only applies to serial ports.
+    // @CopyFieldsFrom: SERIAL1_OPTIONS
+    AP_GROUPINFO("OPT1", 6,  Networking_Periph::Passthru, options1, 0),
+
+    // @Param: OPT2
+    // @DisplayName: Serial Port Options EP2
+    // @Description: Control over UART options for Endpoint 2. Only applies to serial ports.
+    // @CopyFieldsFrom: SERIAL1_OPTIONS
+    AP_GROUPINFO("OPT2", 7,  Networking_Periph::Passthru, options2, 0),
+
+    AP_GROUPEND
+};
+
+void Networking_Periph::Passthru::init()
+{
+    if (enabled == 0) {
+        // Feature is disabled
+        return;
+    }
+
+    if (port1 != nullptr || port2 != nullptr) {
+        // Endpoints have already been initilized
+        return;
+    }
+
+    if (ep1 <= -1 || ep2 <= -1 || ep1 == ep2) {
+        // end points are not set or are the same. Can't route to self
+        return;
+    }
+
+    port1 = AP::serialmanager().get_serial_by_id(ep1);
+    port2 = AP::serialmanager().get_serial_by_id(ep2);
+
+    if (port1 != nullptr && port2 != nullptr) {
+        port1->set_options(options1);
+        port1->begin(baud1);
+
+        port2->set_options(options2);
+        port2->begin(baud2);
+    }
+}
+
+void Networking_Periph::Passthru::update()
+{
+    if (enabled == 0 || port1 == nullptr || port2 == nullptr) {
+        return;
+    }
+
+    // Fastest possible connection is 3Mbps serial port, which is roughly 300KB/s payload and we service this at <= 1kHz
```

Comment:
Hi, How do we come to this number of 3Mbps?

Reply:
UARTs can't really go higher than that reliably. Is there a different number you're thinking? Bigger/smaller?

Reply:
> UARTs can't really go higher than that reliably. Is there a different number you're thinking? Bigger/smaller?

No, I was just wondering if there is any theoretical way to derive it.